### DISCUSSION TITLE
Expectation of global data and events

### DISCUSSION THREAD

**Discussion by cfjedimaster** [11ty/eleventy#2062] [Q&A]
I was playing around with the new add eleventyConfig.addGlobalData feature and noticed you could *not* use it in beforeBuild. It doesn't throw an error, but if a template tries to make use of the variable it won't work. 

Is this expected? To me, I'd expect it to work and create data available to templates.

Why would I use it here? Ok, so my particular use case probably wasn't a practical idea, but this is what I was doing:

```js
	eleventyConfig.on('beforeBuild', () => {
		console.log('beforeBuild');
		eleventyConfig.addGlobalData('test2', Object.keys(eleventyConfig.globalData));
		console.log('done', JSON.stringify(eleventyConfig.globalData));
	});
```

Basically a global data variable that stored the keys of all available other global data variables. I only tried it here as I noticed similar code in the main function only worked right if placed after every other call.
